src/main/java/duke/Parser.java

```diff
@@ -4,32 +4,36 @@
  * This class translate the user input into commands understandable by Duke.
  */
 public class Parser {
-    private TaskList taskList;
-
-    /**
-     * Construct a Parser from the specified list of task.
-     */
-    Parser(TaskList taskList) {
-        this.taskList = taskList;
-    }
 
     /**
      * Parse the user input.
      * @param command input entered by user
      */
-    void parse(String command) {
-        if (command.equals("bye")) {
-            Ui.exit();
-        } else if (command.equals("list")) {
-            taskList.listTask();
-        } else if (command.startsWith("done")) {
-            taskList.markDone(command.substring(5));
-        } else if (command.startsWith("delete")) {
-            taskList.deleteTask(command.substring(7));
-        } else if (command.startsWith("find")) {
-            taskList.findTask(command.substring(5));
-        } else {
-            taskList.addTask(command);
+    public static String parse(String command, TaskList taskList, Storage storage) throws DukeException {
+        command = command.trim();
+        String type = command.substring(0, command.indexOf(" "));
+        String desc = command.substring(command.indexOf(" ") + 1,
+                ((!command.contains("/")) ? command.length() : command.indexOf("/")));
+        String time = command.substring(command.indexOf("/") + 4);
+        String ret;
+        if (type.equals("todo")) {
+            ret = Executor.add(taskList, type, desc);
```

Comment:
Perharps you can use a clearer and more intuitive variable name here! Other than that your coding standards looks good!